### DISCUSSION TITLE
Pick A Keyframe To Be The First Frame

### DISCUSSION THREAD

**Discussion by rd-sf** [mifi/lossless-cut#1259] [Ideas]
Has this been requested?  The ability to stop the timeline on a keyframe, hit a button, and have LC pull that keyframe and copy it to the beginning of the video stream.  This would help folks who use players that use the very first frame as the thumbnail, like iOS does.  Many Many Thanks for making and maintaining Lossless Cut.

**Comment by mifi** [comment]
I beleive "keyframe cut" will effectively do what you're asking, or am I misunderstanding?

**Reply by rd-sf** [reply]
Thank you for writing LosslessCut and taking user feedback.  We owe you a great deal of gratitude. 

Clarification: Many players (like on iOS) create a thumbnail based on the first frame, which is often a fade in from black so the thumbnail is just black. It would be nice to designate one keyframe from anywhere in the file to show up as the very first frame.  I would use this to select the keyframe that shows the date and location of a live Phish concert, which is usually displayed about 10 seconds into the stream.  A more common use case would be to grab a keyframe from TV shows that show the S#E# Title near the beginning but not on the first frame.  Many thanks!

**Reply by rd-sf** [reply]
I have watched your 33 second video "How to how to add cover art" but that's more complex than my request.  This is merely making a segment that is just one frame and putting ahead of everything else.  That one-frame-segment should not be cut-out of the file, but instead copied so that frame occurs in it's natural space in the file AND at the first frame.

**Reply by mifi** [reply]
Ahh, you want to copy an arbitrary frame as the first keyframe, so that when users play back the video, it will flash that first frame for say 1/30 of a second and then start playing the rest?

It's the first time I heard of this feature request, and I would think that being able to select a thumbnail / cover art from an arbitrary frame could solve this use case also? (in a more standard way) see #1100

**Reply by rd-sf** [reply]
Hello, thank you for pointing me to [#1100](https://github.com/mifi/lossless-cut/issues/1100).  I would try it, but create-thumbnail-from-timeline is not a feature yet, correct?

**Reply by mifi** [reply]
correct. you would need to manually take a snapshot, then save it as jpg, then add that screenshot as a cover art

**Reply by rd-sf** [reply]
Correct - goal is to be able to select one keyframe to be cloned and placed as the very first frame in the video file and have it visible for a fraction of a second (your perfect example: 1/30th of a second).  There is no other way to pick a thumbnail in iOS, even if you use MacOS to change the thumbnail of the file in the finder (by doing get-info and pasting an image file from LosslessCut onto the file's thumbnail, essentially replacing it) but that change is not reflected in the video file once it's back on an iOS device.

**Reply by mifi** [reply]
I created a feature request for this #1268

**Comment by rd-sf** [comment]
TY TY
